### ISSUE TITLE
Installation issue: KeyError: 'CPPFLAGS' 

### ISSUE THREAD

**Issue by Huii** [wannesm/dtaidistance#84]
I'm currently again attempting to install dtaidistance on a Windows 10 machine. However, there seems to be an issue with the file setup.py as this is the only message I get:

```
    File "PathTo\dtaidistance\setup.py", line 234, in <module>
        os.environ["CPPFLAGS"] = "-Idtaidistance/lib/DTAIDistanceC/DTAIDistanceC" + " " + os.environ["CPPFLAGS"]
      File "C:\Programs\Anaconda\lib\os.py", line 678, in __getitem__
        raise KeyError(key) from None
    KeyError: 'CPPFLAGS'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

Any help would be highly appreciated as I'm in dire need to run this software for some comparison studies. Thank you so much.

**Comment by Huii** [comment]
I was able to "solve" this issue by changing line 234 of the file setup.py from

`    os.environ["CPPFLAGS"] = "-Idtaidistance/lib/DTAIDistanceC/DTAIDistanceC" + " " + os.environ["CPPFLAGS"]`
to
`    os.environ["CPPFLAGS"] = "-Idtaidistance/lib/DTAIDistanceC/DTAIDistanceC" + " " #+ os.environ["CPPFLAGS"]`

However, I'm not sure that this won't break anything else in the future.

**Comment by Huii** [comment]
After that, I ran into the good old "AttributeError: 'MSVCCompiler' object has no attribute 'compiler'" issue again (see https://github.com/wannesm/dtaidistance/issues/43). I solved it by simply changing the line

`check_result = check_openmp(self.compiler.compiler[0])`

in the file setup.py to:

`check_result = True`

Unfortunately, I'm now stuck with yet another error message:

`dtaidistance/dtw_cc.c(627): fatal error C1083: File (Include) cannot be opened: "lib/DTAIDistanceC/DTAIDistanceC/globals.h": No such file or directory`

Any ideas?

**Comment by fuchaoqun** [comment]
same on ubuntu 20.04 with anaconda

pip install dtaidistance==2.0.0
Collecting dtaidistance==2.0.0
  Using cached dtaidistance-2.0.0.tar.gz (1.3 MB)
    ERROR: Command errored out with exit status 1:
     command: /opt/anaconda3/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-1ei54ypj/dtaidistance/setup.py'"'"'; __file__='"'"'/tmp/pip-install-1ei54ypj/dtaidistance/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-1ei54ypj/dtaidistance/pip-egg-info
         cwd: /tmp/pip-install-1ei54ypj/dtaidistance/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-1ei54ypj/dtaidistance/setup.py", line 234, in <module>
        os.environ["CPPFLAGS"] = "-Idtaidistance/lib/DTAIDistanceC/DTAIDistanceC" + " " + os.environ["CPPFLAGS"]
      File "/opt/anaconda3/lib/python3.7/os.py", line 679, in __getitem__
        raise KeyError(key) from None
    KeyError: 'CPPFLAGS'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

**Comment by wannesm** [comment]
Thanks for the detailed feedback. I overlooked Windows. The setup.py file is now restructured to have the additional paths in the top of the file where the platform customization is done (e.g. it should be using \I instead of -i for msvc). I cannot test this easily for different platforms, if you want to expect the path, see line setup.py:39.

The Ubuntu error was due to a wrong use of os.environ. That's now also moved to the compiler settings and should be fixed.

The source distribution was also forgetting about the pxd files. That's also fixed.

A new version is pushed to github and pipy.
